### DISCUSSION TITLE
Different Memory size executables

### DISCUSSION THREAD

**Discussion by krishna3008** [pyinstaller/pyinstaller#5949] [PyInstaller]
Hi
I created a python exe using pyinstaller and the memory size of the exe is around 132MB and same file ran in another PC which created a file of size 63MB and in another PC 340MB

Can you please inform why this variation and how to create a exe with lowest possible size

**Comment by rokm** [comment]
Well, from the amount of information you provided, our guess is as good as yours. But it probably boils down to having different environments on all three systems, e.g., different (optional) packages installed in the (virtual) python environment. So to minimize the executable size, you should probably create a new, clean virtual environment, and install only the packages your program really requires, in order to prevent optional imports from being pulled in.
